File: python/11. Container With Most Water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        res = 0
        l, h = 0, len(height) - 1
        while l < h:
            area = 0
            if height[l] < height[h]:
                area = (h-l)*height[l]
                l += 1
            else:
                area = (h-l)*height[h]
                h -= 1
            res = max(area, res)
        
        return res
        

File: go/src/leetcode/11. Container With Most Water/11. Container With Most Water.go
package main

import (
	"math"
)

func maxArea(height []int) int {
	l, r := 0, len(height)-1
	maxVal := 0
	for l < r {
		maxVal = max(maxVal, (r-l)*min(height[l], height[r]))
		if height[r] > height[l] {
			l++
		} else {
			r--
		}
	}
	return maxVal
}

func max(a, b int) int {
	return int(math.Max(float64(a), float64(b)))
}

func min(a, b int) int {
	return int(math.Min(float64(a), float64(b)))
}
